apps/charterafrica/src/lib/tools/models.js

```diff
@@ -0,0 +1,132 @@
+import { FetchError } from "../../utils/fetchJson";
+import api from "../payload";
+
+export const ORGANIZATION_COLLECTION = "organisations";
+export const PEOPLE_COLLECTION = "people";
+export const TOOL_COLLECTION = "tools";
+export const DIGITAL_DEMOCRACY_ECOSYSTEM = "digital-democracy-ecosystem-config";
+
+function removeDuplicatesByKey(arr, key) {
+  const uniqueKeys = new Set();
```

Comment:
Since we want to deduplicate based on key, I think a Map is better since it retains [insertion order](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) as well
```js
const uniqueItemsByKey = arr?.reduce((acc, curr) => {
  const keyValue = curr?.[key]
  if(keyValue) {
    acc[keyValue] = curr; // Will override if keyValue already exists
  }
  return acc;
  }, new Map());
return uniqueItemsByKey?.values() ?? [];
````

Haven't tested the code though